src/caf/distribute/gravity_model/multi_area.py

```diff
@@ -243,35 +372,67 @@ def _create_seed_matrix(self, cost_distributions, cost_args, params_len):
     # pylint: disable=too-many-locals
     def _calibrate(
         self,
+        distributions: MultiCostDistribution,
+        running_log_path: Path,
+        furness_jac: bool = False,
         diff_step: float = 1e-8,
         ftol: float = 1e-4,
         xtol: float = 1e-4,
+        furness_tol=DEFAULT_FURNESS_TOL,
         grav_max_iters: int = 100,
         failure_tol: float = 0,
         default_retry: bool = True,
         verbose: int = 0,
         **kwargs,
-    ) -> dict[str, GravityModelCalibrateResults]:
-        params_len = len(self.dists[0].function_params)
+    ) -> dict[str | int, GravityModelCalibrateResults]:
+        params_len = len(distributions[0].function_params)
         ordered_init_params = []
-        for dist in self.dists:
+        # TODO validate cost distributions
```

Comment:
I think it would be cleaner if the validation of cost distributions was moved to another private method to keep calibrate simpler.